cql3/statements/select_statement.cc

```diff
@@ -1081,22 +1081,24 @@ lw_shared_ptr<const service::pager::paging_state> indexed_table_select_statement
     auto& last_base_pk = last_pos.partition;
     auto* last_base_ck = last_pos.position.has_key() ? &last_pos.position.key() : nullptr;
 
-    bytes_opt indexed_column_value = restrictions::value_for(*cdef, _used_index_restrictions, options);
+    std::vector<bytes_opt> indexed_column_vector = restrictions::value_for(*cdef, _used_index_restrictions, options);
 
     auto index_pk = [&]() {
         if (_index.metadata().local()) {
             return last_base_pk;
         } else {
-            return partition_key::from_single_value(*_view_schema, *indexed_column_value);
+            return partition_key::from_single_value(*_view_schema, *indexed_column_vector.back());
```

Comment:
Let's add value_list_for that returns a vector, and change value_for() to use list_value_for() and verify it has exactly one element, avoiding the chance that we reference an empty vector or ignore some elements. It probably can't happen due to grammar constraints, but let's be defensive.

Reply:
I've designed value_for to return a vector of possible values for any operation on RHS, particularly IN operation.
For example, in "SELECT ck FROM tb WHERE x IN (2, 4)" query possible values would be 2 and 4. I don't see why we should check that value_for returns a single element. But, i agree that we could check if vector is empty to prevent errors.